tests/migration/check_system_registration.pm

```diff
@@ -0,0 +1,34 @@
+# SLE15 migration tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Register the original system before migration
```

Comment:
The code here should be more straightforward, without conditions.

For migrations in maintenance we don't need register the system again, it is already connected with a previous installation of SLE 12 SP5 which wasn't deregistered.

Reply:
The system is registered the first time the job runs, but if there is a failure and we need to re-run the job, the system is not registered.

Reply:
the image is stored registered, the first child job is using a copy of that image, the second time the child job run will be using another copy, but still registered, the original published image is untouched. Could you probe that with a parent/child links showing that error?

Reply:
actually, perhaps the child job during upgrade via YaST and YaST calling to SCC might do something with the registration, so the next run doesn't work ... I see a potential catch you found it ... I would like to see it to be sure (you don't need AutoYaST to share this).

Reply:
Yes, there already are a lot of relevant failures. example for autoyast installation failure: https://openqa.suse.de/tests/13933716#step/installation/4 I was also suspecting that an unsuccessful migration might cancel the product registration on scc. I will ask in the channel

Reply:
Basically, I have added a record_info for checking if the system is already registered: 
Already registered: https://openqa.suse.de/tests/13929395#step/check_system_registration/10
Not registered (rerun after failure): https://openqa.suse.de/tests/13934183#step/check_system_registration/10
Not registered (rerun after successful migration): https://openqa.suse.de/tests/13943135#step/check_system_registration/10

Reply:
makes sense it cannot be reused, again great catch here @sofiasyria ! :)

So to have fixed expectation we should not have this kind of modules that do something depending on what they find, so we need to unregister always in parent and register always in child.
The only way to do that in the way I thought was with START_DIRECTLY_AFTER, because force you to use the same vm and restart both, but that is not needed, we might need to soft-fail a migration because something is not fixed by developers fast enough and still want to test the installation. Migration are going to be secondary until we know if they are stable. So yeah, let's do it as you said, which is the way is also in product validation forcing de-registration and registration, will be easy to restart only the child to debug problems.
cc @hjluo

Reply:
Ok, just a clarification. I will deregister and reregister the system during the migration test, correct? I will not touch the parent test that publishes the qcow

Reply:
no, the deregistration we usually do in the parent, so we save time in the child, the child just need to register always unconditionally. Basically it is what we know that always we store a qcow2 in openqa needs to be unregistered due to scc metadata remaining in the image published which is used onward.

Reply:
well noticed, then we can unconditionally unregister and register again in the child.